hibernate-core/src/main/java/org/hibernate/cfg/AnnotationBinder.java

```diff
@@ -3163,7 +3163,9 @@ else if ( oneToOne != null ) {
 		}
 		else {
 			toOne.setLazy( fetchType == FetchType.LAZY );
-			toOne.setUnwrapProxy( false );
+
```

Comment:
Should the following be set to `true` when ` fetchType != FetchType.LAZY`

Will this change behavior for non-lazy attributes that end up getting loaded as a proxy from the `PersistenceContext`?

I'd like to look more deeply at this on Monday.

Reply:
I think yes. I tried, and it fixed the bug ;)

Reply:
I think that is probably true

Reply:
This is fine.

Reply:
What I found is that knowing when the LazyToOne was implicit is just as important.  To that end, this is now as follows in my newest work, which I'll be pushing as a new PR soon -

```
toOne.setUnwrapProxy( fetchType != FetchType.LAZY );
toOne.setUnwrapProxyImplicit( true );
```

this new "implicitness" is checked later.  Ideally unwrap-proxy would have been a Boolean value allowing null rather than a boolean forcing a true/false here.  But I felt adding a new method was less disruptive since that mapping model is considered an API